### PR TITLE
mlir attr types for infrt place (1st PR), test=develop

### PR DESCRIPTION
**Pull request by Shixiaowei02**
### PR types

New features
### PR changes

Others
### Describe

1. add infrt base types and attributes for precision, layout, and target.
2. apply types to the context and allocator.
3. use generated codes for parsing and printing whenever possible.

### CODE REVIEW

**Review by Superjomn — DISMISSED** [review]
LGTM

File: paddle/infrt/dialect/infrt/CMakeLists.txt [hunk 1]
```diff
@@ -8,6 +8,12 @@ gather_srcs(infrt_src SRCS
 
 add_mlir_dialect(infrt_ops infrt)
 
+set(LLVM_TARGET_DEFINITIONS infrt_ops.td)
```
Comment (winter-wang) [inline paddle/infrt/dialect/infrt/CMakeLists.txt]: 感觉新加的这五行代码和随后的五行代码功能是一模一样的啊，。。这几行代码是不是没必要？

Comment (Shixiaowei02) [reply paddle/infrt/dialect/infrt/CMakeLists.txt]: fixed, thanks.

**Review by winter-wang — APPROVED** [review]
LGTM

**Pull request merged**